src/extension/notifier/src/main/java/au/org/emii/wps/NotifierProcess.java

```diff
@@ -39,8 +43,31 @@ public RawData execute(
     ) throws ProcessException {
 
         try {
-            httpNotifier.notify(callbackUrl, getWpsUrl(), getId(), callbackParams);
-            return notifiableData;
+            IOException lastException;
```

Comment:
I think I'd rather see this logic go in the `HttpNotifier` class, e.g. an overloaded method `notify()`, which also takes as parameters the `maxNotificationAttempts` and `retryInterval`.  This class then remains as just "plumbing" of the interface between geoserver/WPS and our custom code.